ui-app/client/src/components/FormStorageEditor/components/formCommons.ts

```diff
@@ -0,0 +1,586 @@
+import { ComponentDefinition } from '../../../types/common';
+
+interface ItemSchema {
+	type: Array<string>;
+}
+export interface CustomSchema {
+	type?: Array<string>;
+	enums?: Array<string>;
+	pattern?: string;
+	maxLength?: number | undefined;
+	minLength?: number | undefined;
+	minimum?: number | undefined;
+	maximum?: number | undefined;
+	items?: ItemSchema;
+	oneOf?: Array<CustomSchema>;
+	format?: string;
+}
+
+export interface FormSchemaProperties {
+	[key: string]: CustomSchema;
+}
+
+export interface FormSchema {
+	type: 'OBJECT';
+	name?: string;
+	properties?: FormSchemaProperties;
+	required?: Array<string>;
+	additionalProperties: false;
```

Comment:
Remove this property and default it to false in the actual schema. Usually we don't use literals as types unless we have fixed number of multiple literals.